backend/model/User.ts

```diff
@@ -0,0 +1,18 @@
+import mongoose, { Schema, model, connect, Types } from 'mongoose';
+
+export interface IUser extends Document {
+    fullName: string;
+    email: string;
+    photoUrl: string;
+    roles: string;
+  }
+
+  const UserSchema: Schema = new Schema({
+    fullName: {type: 'string', required: true},
+    email: {type: 'string', required: true},
+    photoUrl: {type: 'string', required: true},
+    roles: {type: 'string', required: true},
```

Comment:
```suggestion
     role: {
      type: String,
      enum: {
        values: ["student", "teacher"],
        message: "Role must be student, or teacher",
      },
      default: "user",
    }
```
Since `role` should only be `student`, or `teacher`, we should make it an enum